impeller/renderer/backend/metal/context_mtl.mm

```diff
@@ -377,17 +377,21 @@ new ContextMTL(device, command_queue,
   return buffer;
 }
 
-void ContextMTL::StoreTaskForGPU(const std::function<void()>& task) {
-  tasks_awaiting_gpu_.emplace_back(task);
+void ContextMTL::StoreTaskForGPU(const std::function<void()>& task,
+                                 const std::function<void()>& failure) {
+  tasks_awaiting_gpu_.push_back(PendingTasks{task, failure});
   while (tasks_awaiting_gpu_.size() > kMaxTasksAwaitingGPU) {
-    tasks_awaiting_gpu_.front()();
+    PendingTasks front = std::move(tasks_awaiting_gpu_.front());
+    if (front.failure) {
+      front.failure();
+    }
     tasks_awaiting_gpu_.pop_front();
   }
 }
 
 void ContextMTL::FlushTasksAwaitingGPU() {
   for (const auto& task : tasks_awaiting_gpu_) {
-    task();
+    task.task();
```

Comment:
Perform null checks here or wrap a null ptr in a do-nothing fml::closure in `StoreTaskForCPU` perhaps?